pyrsa/data/dataset.py

```diff
@@ -137,6 +139,20 @@ def subset_channel(self, by, value):
         """
         raise NotImplementedError(
             "subset_channel function not implemented in used Dataset class!")
+    def save(self, filename):
+        """ Saves the dataset object to two files:
+            [filename].npy : the data in numpy format
+            [filename]_desc.pkl : pickle of the descriptors
```

Comment:
Have you considered using hdf5? Then you could store the descriptors as a dictionary or whatever in the same file. (So the user does not get confused about the two files). Also hdf5 is a bit more cross-platform (e.g. matlab).

Reply:
Thanks for this comment. I am quite unsure what the best format is for saving here. I'll try out hdf5 this afternoon. If it can save all the content this would be better I think

Reply:
hmmm hdf5 seems to imply limitations to the descriptors as we would have to save them in the attribute parts. Dictionaries cannot be written directly to hdf5 apparently. I guess I could usually convert them such that they form groups which only contain strings and arrays, such that I can save them, but that honestly sounds like a bit of work to me.

I am somewhat tempted to just convert the object to a dictionary of dictionaries and np.arrays and pickle that as a whole. Do you know how wasteful this is compared to saving the main part of the object (i.e. the RDMs/measurements/evaluations) as hdf5 data and the rest as attribute annotations? Or is there any other reason to prefer hdf5 over the pickle, if I save only NumPy arrays and strings within layered dictionaries?

Reply:
never mind. I have now started the conversions to dictionaries. I'll then write a function which writes dictionaries into pickles and one which writes them into hdf5. That should expose the choice to the user and if they used something which cannot be saved to hdf5 this will raise an error. That seems fair to me. Any objections?

Reply:
now implemented as described